src/shell_minimal.js

```diff
@@ -0,0 +1,62 @@
+// Copyright 2010 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#if SIDE_MODULE == 0
+#if USE_CLOSURE_COMPILER
+// if (!Module)` is crucial for Closure Compiler here as it will otherwise replace every `Module` occurrence with a string
+var Module;
+if (!Module) Module = "__EMSCRIPTEN_PRIVATE_MODULE_EXPORT_NAME_SUBSTITUTION__";
+#else
+var Module = {{{ EXPORT_NAME }}};
+#endif // USE_CLOSURE_COMPILER
+#endif // SIDE_MODULE
+
+// Redefine these in a --pre-js to override behavior
+function out(text) {
+  console.log(text);
+}
+
+function err(text) {
+  console.error(text);
+}
+
+function ready() {
+  run();
+}
+// --pre-jses are emitted after the Module integration code, so that they can
+// refer to Module (if they choose; they can also define Module)
+// {{PRE_JSES}}
```

Comment:
In MINIMAL_RUNTIME, one customizes stdout/stderr printing and automatic vs manual startup by overriding `out`, `err`, or `ready` in a --pre-js file. Closure can optimize the pattern

```js
function out(text) {
  console.log(text);
}

...

function out(text) {
  document.getElementById('my_console_out').value += text;
}
```

to remove the first function. Also, Closure is able to optimize the pattern

```js
function out(text) {
  console.log(text);
}

...

function out(text) {
  /* no-op*/
}
```

to take away all calls to `out()` in call sites, so this is a nice way to delete away default logging functionality.

One can override `ready()` to get a signal of when the runtime is ready, and then either choose to run the code immediately, or call `run()` later when one wants to.

Reply:
Worth mentioning the closure benefits in a comment in the code, I think.